### ISSUE TITLE
[Feature Proposal] Cross-Platform error/info logging for Android

### ISSUE THREAD

**Issue by Str4t0n** [nnstreamer/nntrainer#685]
Feature proposal for a Cross-Platform logging header/option to be used inside NNTrainer.

**Reason**: If nntrainer crashes inside of an android project, there will be no error log (nor any information of the crash), because inside an android project, the cerr/cout standard output stream is not working/valid.

Something like this:
```cpp
#ifdef ANDROID
    // Logs for Android debugging / info
    #include <android/log.h>
    #define LOGV(tag, ...) __android_log_print(ANDROID_LOG_VERBOSE, tag, __VA_ARGS__)
    #define LOGD(tag, ...) __android_log_print(ANDROID_LOG_DEBUG  , tag, __VA_ARGS__)
    #define LOGI(tag, ...) __android_log_print(ANDROID_LOG_INFO   , tag, __VA_ARGS__)
    #define LOGW(tag, ...) __android_log_print(ANDROID_LOG_WARN   , tag, __VA_ARGS__)
    #define LOGE(tag, ...) __android_log_print(ANDROID_LOG_ERROR  , tag, __VA_ARGS__)
    #define LOGSIMPLE(...)
#else
    // Normal logging
    #include <stdio.h>
    #define LOGV(tag, ...) printf("\t - V/%s: ", tag); printf(__VA_ARGS__); printf("\n");
    #define LOGD(tag, ...) printf("\t - D/%s: ", tag); printf(__VA_ARGS__); printf("\n");
    #define LOGI(tag, ...) printf("\t - I/%s: ", tag); printf(__VA_ARGS__); printf("\n");
    #define LOGW(tag, ...) printf("\t - W/%s: ", tag); printf(__VA_ARGS__); printf("\n");
    #define LOGE(tag, ...) printf("\t - E/%s: ", tag); printf(__VA_ARGS__); printf("\n");
    #define LOGSIMPLE(...) printf("\t Info: ", );printf(__VA_ARGS__);
#endif // ANDROID
```

**Comment by myungjoo** [comment]
You may update the code and send Pull-Requests; that'd be the fastest method. :)

**Comment by zhoonit** [comment]
Seems like this was already implemented in c218e63 , closing this
